### ISSUE TITLE
First screenshot png repeated 

### ISSUE THREAD

**Issue by MatteoRomiti** [kevinhughes27/TensorKart#27]
I installed all the dependencies and everything works fine, except the recording: I get always the same initial screenshot png repeated over and over. Joystick inputs are displayed in the graph and recorded correctly. 
Any suggestions?

OS: Ubuntu 16.04

**Comment by bzier** [comment]
For the screenshot you do get, is it a good screenshot of the emulator window, or is it just a black window, or some other snippet from your screen? Does the preview in the recording window update, or is that static the whole time too?

**Comment by MatteoRomiti** [comment]
It is a good screenshot of the emulator and the preview in the recording window does not update the image, it updates only the joystick input.

**Comment by kevinhughes27** [comment]
That's super weird - what are the rest of your system specs (video card, video driver and desktop env?). I'd recommend playing around with this code here: https://github.com/kevinhughes27/TensorKart/blob/master/utils.py#L24

**Comment by MatteoRomiti** [comment]
I have a NVIDIA Geforce GTX 950M. I had the NVIDIA graphics drivers-367, then I tried with version 375 and finally with 378, but nothing changed unfortunately. I'll keep playing with the code you suggested and maybe I'll try another way to take screenshots (suggestions are welcome).

Desktop environment: GNOME - Unity

**Comment by MatteoRomiti** [comment]
So, I found a workaround that does the job, probably a bit slower but acceptable. In case someone has the same problem, you can find the alternative implementation on my fork. 
@kevinhughes27 please let me know if you want me to do/correct something.

P. S. 
I don't understand something here: https://github.com/kevinhughes27/TensorKart/blob/master/play.py#L69
Why are we sending a list instead of a JSON? I changed it to the JSON format as in https://github.com/kevinhughes27/mupen64plus-input-bot/blob/master/test/server.py#L36

**Comment by bzier** [comment]
In response to your P.S.:
That was something that Kevin and I changed last weekend in the input bot. I guess neither one of us got around to updating this project to match. Originally, the input bot was just parsing string data for each of the controller values (splitting the list on commas). We updated it to be structured with JSON and added support for each of the controller buttons (not just A, B, RB and the joystick axes). The sample server you referenced is a good example of the new format.
I may get around to updating this project today, so if I do, I'll submit a pull request and hopefully Kevin can get it merged in. Sorry I broke it for you, but glad you got it working.
